src/axolotl/utils/models.py

```diff
@@ -81,6 +82,16 @@ def load_model(
         logging.exception(e)
         raise e
 
```

Comment:
We need to also
- assert that both `cfg.load_in_8bit` and `qlora` and `load_4bit` are not True. Only one can be True at a time.
- Fix the `torch_dtype` since it does not factor in for qlora 4bit.
- Would `bf16` affect the training? Looking at the sample fine-tuning code, they used `fp16`.

(I cannot reference those lines in the review UI)

Reply:
I addressed #1, I didn't see anything that needed to be addressed for #2 (was there something else you caught that I missed?) I think bf16 would be better for the parameters that use that since it would avoid potential overflows.